### ISSUE TITLE
Error copying files with ’ (quote) character in them on OS X 10.10

### ISSUE THREAD

**Issue by deepakjois** [s3tools/s3cmd#526]
I am running s3cmd on OS X 10.10

```
$ s3cmd cp Kiran\ Ahluwalia’s\ Tracks s3://www.deepak.jois.name
ERROR: Parameter problem: Expecting S3 URI instead of 'Kiran Ahluwalia���s Tracks'
```

**Comment by mdomsch** [comment]
You have to escape (\') the apostrophe so the shell doesn't get confused.

**Comment by deepakjois** [comment]
Actually, this does not work either.

```
s3cmd cp "Kiran Ahluwalia’s Tracks" s3://www.deepak.jois.name
```

Moreover, I have a similar problem when I do `s3cmd sync` on a folder containing files that are named similarly. It appears to be a problem with the ‘type’ of the S3 URI as detected by s3cmd’s S3URI class.

**Comment by mdomsch** [comment]
You don't mention the version of s3cmd you are using.  With upstream master
branch (slightly newer than v1.5.2), it works for me...

$ ./s3cmd put ../s3cmd-tests/file-with-\'-in-the-name s3://
s3cmd-test.domsch.com/
../s3cmd-tests/file-with-'-in-the-name -> s3://
s3cmd-test.domsch.com/file-with-'-in-the-name  [1 of 1]
 20 of 20   100% in    0s    97.62 B/s  done

$ ./s3cmd sync ../s3cmd-tests/Kiran\ Ahluwalia▒\▒▒s\ Tracks  s3://
s3cmd-test.domsch.com/
../s3cmd-tests/Kiran Ahluwalia’s Tracks/music.txt -> s3://
s3cmd-test.domsch.com/Kiran Ahluwalia’s Tracks/music.txt  [1 of 1]
 23 of 23   100% in    0s   152.63 B/s  done
Done. Uploaded 23 bytes in 1.0 seconds, 23.00 B/s. Copied 0 files saving 0
bytes transfer.

Thanks,
Matt

**Comment by deepakjois** [comment]
The version I am running is v1.5.2, on OS X 10.10 using iTerm2.

I could try with the latest version of master, but I don’t really have an environment ready for Python dev. Feel free to close if you believe this is either a problem with my environment, or something that has already been fixed.

I will report back if I find out something.

**Comment by fviard** [comment]
This issue is very very old, but in case someone arrive here, there is a simple explanation.

@deepakjois You were confused on the command to use.

`s3cmd cp` is for a "remote copy", ie a copy from s3 to s3.
So, your arguments should have been: 
`s3cmd cp s3://bucket1/file1 s3://bucket2/file2.`
But you used a local file, that you wanted to send to the cloud.
So, depending of your need, you should have used: `s3cmd put` or `s3cmd sync`.

In conclusion, there was no "quote" issue, and the error message was right: "Source should have been a s3 address".
